.github/workflows/lint.yml

```diff
@@ -0,0 +1,25 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: FxA Lint
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version-file: '.nvmrc'
+        cache: 'yarn'
+    - run: yarn install --frozen-lockfile
+    - run: npx eslint . --config _dev/.eslintrc --ext .js
```

Comment:
Instead of using a global, shared config (which is what husky is using IIRC), should we just use something like this?
```sh
yarn workspaces foreach run lint
```

I think that will call each workspace's "lint" script in its package.json file, which means we'd be linting JavaScript/TypeScript/CSS/Sass/whatever for free. And then each project still can maintain it's own config/rules separately.

Reply:
I don't know.  However people lint today works for me.  I'm a little wary of each project maintaining its own rules, tbh.

Reply:
> However people lint today works for me.

I've been relying on the pre-commit hook fwiw.  I'm probably relying on the per package configs while I'm in VS Code, but I rarely manually run 'yarn lint' in a package.

Reply:
I tend to agree with what Peter suggested. I think it’s inevitable that there will be eslint plugins and rules that have been added that specifically address some sort of linting concern in a package. For example a react app with typescript probably needs different eslint plugins/rules than a node server with vanilla JavaScript.
 
On that note, eslint should be running on more than just .js files, so the --ext argument should probably be extended to include .ts and .tsx files. And just a random nitpick, declaring --ext .js is unnecessary because this is already the default behavior.

Reply:
nitpicks are fine - I don't know how we normally run linting.  What should the final command be?  When I run `yarn workspaces foreach run lint` locally it fails looking for configs.  Should that be working?

Reply:
> Should that be working?

Definitely. I think it means we have repos that aren't really being 100% linted, apart from whatever minimal rules are enforced by the shared [_dev/.eslintrc](https://github.com/mozilla/fxa/blob/main/_dev/.eslintrc) config file.
Currently .ts or .tsx files aren't linted at all (only formatted via Prettier), per https://github.com/mozilla/fxa/blob/893226413f8aec7edb018644d295620d79b3b4b1/package.json#L60-L62

Potentially related:
- https://github.com/mozilla/fxa/issues/11718
- https://github.com/mozilla/fxa/issues/11732
- https://github.com/mozilla/fxa/issues/11762

Not sure how we'd globally lint .ts files without letting each repo define it's own config via a https://github.com/mozilla/fxa/blob/d1283314c15d81a85f08dcba5bce329db7c6fd51/packages/fxa-content-server/tsconfig.json file. But that also might depend on what rules are defined.

Reply:
> I'm a little wary of each project maintaining its own rules, tbh.

I think that's always been the way. Back in the early-early days we had to special case rules in the eslint-config-fxa package w special configs for content-vs-auth "quotes" and "semicolons" because some repos used single/double quotes, and others repos were strongly opposed to semicolon usage. Not sure of a workaround apart from refactoring everything to TypeScript and having one-true-config to rule them all and reformatting the entire repo to use a single style guide (which I'm not opposed to, but it might be a lot of work/noise).

Reply:
I fixed the paths and ran the linter and about half the packages are fine but others (👀 at you content-server) have hundreds of errors.  So, yeah, running linting in ci might be premature...

Reply:
Ok, I'm going to hit resolve on this because I pushed a new patch using the new command.